src/Iterators.jl

```diff
@@ -33,9 +33,21 @@ if VERSION < v"0.5.0-dev+3305"
     function iteratorsize(v)
         error("Do not call this on older versions")
     end
+
+    has_length(it) = applicable(length, it)
 else
     import Base: iteratorsize, SizeUnknown, IsInfinite,
                 HasLength, HasShape
+
+    function has_length(it)
+        it_size = iteratorsize(it)
+
+        if isa(it_size, HasLength) || isa(it_size, HasShape)
+            return true
+        else
+            return false
+        end
```

Comment:
Hah, maybe I should have waiting until tomorrow to submit!

I'll change that with a rebase.